src/main/kotlin/camp/nextstep/edu/step2/StringCalculator.kt

```diff
@@ -39,11 +39,6 @@ object StringCalculator {
     }
 
     private fun doInfix(left: Double, operator: Operator, right: Double): Double {
-        return when (operator) {
-            Operator.PLUS -> (left.plus(right))
-            Operator.MINUS -> (left.minus(right))
-            Operator.TIMES -> (left.times(right))
-            Operator.DIVIDE -> (left.div(right))
-        }
+        return operator.operate(left, right)
```

Comment:
Nit: 연산자 오버로딩을 이용하면 바로 함수로서 사용도 가능해집니다.
```suggestion
        return operate(left, right) // Operator 객체의 변수명을 operate 로 받았음
```

참고: https://wooooooak.github.io/kotlin/2019/03/21/kotlin_invoke/